### ISSUE TITLE
Inquiry about error message

### ISSUE THREAD

**Issue by simonwisnovsky** [fulcrumgenomics/guide-counter#20]
My lab has recently switched from Mageck to guide-counter for generating count tables, we find it a faster and more accurate alternative. We've processed a number of screens without problems. However, I've recently been trying to re-process some older data and keep getting this error message that we have not seen before, see command below:

(base) guide-counter count -N 100  --input Sig9_C_R1_RC.fastq --library CRISPRilib-20-Top5.txt --output Sig9_C1_R1 
[2023-10-27T18:36:57Z INFO  guide_counter::guide] Loaded library with 103072 guides for 18787 genes; 0=essential, 0=nonessential, 0=control, 103072=other.
[2023-10-27T18:36:57Z INFO  guide_counter::commands::count] Building lookup.
[2023-10-27T18:36:58Z INFO  guide_counter::commands::count] Lookup built with 6285562 entries.
thread 'main' panicked at 'range end index 20 out of range for slice of length 19', src/commands/count.rs:258:38

Any ideas as to what is going wrong here? The only difference between these fastq files and ones we have previously used successfully is that the read length is slightly shorter (21 bp vs. 25 bp). Otherwise conditions are exactly the same.

Thanks in advance!

**Comment by tfenne** [comment]
Thanks for the report @simonwisnovsky - I took a brief look at the code and it's not entirely obvious to me why this is occurring.

Would you be able to share some of your input data with me to help me diagnose?  I suspect that the library file and the first few hundred lines of the input FASTQ would be sufficient (if that reproduces the error).  I can provide contact info if you don't want to share them publicly but can share privately.

Also, can you please confirm you're using v1.0.3?

**Comment by simonwisnovsky** [comment]
Hi Tim,
Of course, thanks very much. I've put a link below to one of the Fastq
files and our library file, please let me know if that works. The library
file has worked fine for other fastq files.
https://www.dropbox.com/scl/fo/t0qgp17r2n3fbaeoxhwzq/h?rlkey=lde99q0lsrcw3kx4ahwflob61&dl=0
Cheers,
Simon
*Prof. Simon Wisnovsky* PhD (He, Him, His
<https://equity.ubc.ca/resources/gender-diversity/pronouns/>)
Assistant Professor
Faculty of Pharmaceutical Sciences
University of British Columbia | Vancouver Campus | Musqueam Traditional
Territory
2405 Wesbrook Mall | Vancouver BC | V6T 1Z3 Canada
***@***.***

**Comment by simonwisnovsky** [comment]
Hi Tim,
Ah ok thanks - is the 0.1.3 version on bioconda? I have run conda update
and it tells me I already have the latest version. I believe I also did the
initial installation after March 2022.
Cheers,
Simon
*Prof. Simon Wisnovsky* PhD (He, Him, His
<https://equity.ubc.ca/resources/gender-diversity/pronouns/>)
Assistant Professor
Faculty of Pharmaceutical Sciences
University of British Columbia | Vancouver Campus | Musqueam Traditional
Territory
2405 Wesbrook Mall | Vancouver BC | V6T 1Z3 Canada
***@***.***

**Comment by tfenne** [comment]
Yes - version 0.1.3 is on bioconda (see https://anaconda.org/bioconda/guide-counter).  I'm not an expert on `conda update`, but I think if you run `conda install -c bioconda guide-counter=0.1.3` directly it should force the update.
